File: Happy.Domain/Entities/Exercise.cs
namespace Happy.Domain.Entities;

public class Exercise : Base<long>
{
    public string Name { get; set; }
    public string Description { get; set; }
    public int? Weight { get; set; }
    public int NumberOfRepetitions { get; set; }

    #region Navigation properties

    public ICollection<ComplexExercise> ComplexExercises { get; set; } = new List<ComplexExercise>();

    public ICollection<Progress> Progresses { get; set; } = new List<Progress>();

    #endregion
}

File: Happy.Client/src/app/models/exercise.ts
export interface IExercise {
    id: number;
    name: string;
    description: string;
    weight?: number;
    numberOfRepetitions?: number;
}

export class Exercise implements IExercise {
    id: number;
    name: string;
    description: string;
    weight?: number;
    numberOfRepetitions?: number;

    constructor(init: ExerciseFormValues) {
        this.id = init.id;
        this.name = init.name;
        this.description = init.description;
        this.weight = init.weight;
        this.numberOfRepetitions = init.numberOfRepetitions;
    }
}

export class ExerciseFormValues {
    id: number = 0;
    name: string = '';
    description: string = '';
    weight?: number = undefined;
    numberOfRepetitions?: number = undefined;

    constructor(init?: ExerciseFormValues) {
        if(init){
        this.id = init.id;
        this.name = init.name;
        this.description = init.description;
        this.weight = init.weight;
        this.numberOfRepetitions = init.numberOfRepetitions;
        }
    }
}